src/sql/workbench/parts/jobManagement/browser/notebooksView.component.ts

```diff
@@ -981,4 +983,26 @@ export class NotebooksViewComponent extends JobManagementView implements OnInit,
 		let ownerUri: string = this._commonService.connectionManagementService.connectionInfo.ownerUri;
 		await this._commandService.executeCommand('agent.openNotebookDialog', ownerUri);
 	}
+
+	public async openLatestNotebookRun(notebook: azdata.AgentNotebookInfo) {
+		let notebookHistories = this._notebookCacheObject.getNotebookHistory(notebook.jobId);
```

Comment:
Why is this async? You aren't awaiting anything. Instead of using .then I'd suggest just turning the below code into :

```
const result = await this._jobManagementService.getMaterialziedNotebook(ownerUri, targetDatabase, history.materializedNotebookId);
if(result)
...
```

so that it behaves like the other actions here and doesn't return until the command has finished executing.

Reply:
Changed to await